src/components/utils.js

```diff
@@ -1,39 +1,53 @@
-const crypto = require('crypto');
+
+import EncryptionUtils from "@verida/encryption-utils"
 
 class Utils {
-  generateUsernameFromRequest(req) {
-    let did = req.auth.user.toLowerCase();
-    let applicationName = req.headers['application-name'];
-    return this.generateUsername(did, applicationName);
-  }
-
-  generateUsername(did, applicationName) {
-    let hash = crypto.createHmac('sha256', process.env.HASH_KEY);
-    hash.update(did + '/' + applicationName);
-    const username = hash.digest('hex');
-
-    // Username must start with a letter
-    return 'v' + username;
-  }
-
-  didsToUsernames(dids, applicationName) {
-    if (!dids || !dids.length) {
-      return [];
+
+    generateUsername(did, contextName) {
+        did = did.toLowerCase()
+        const text = [
+            did,
+            contextName
+        ].join('/')
+
+        const hash = EncryptionUtils.hash(text).substring(2)
+
+        // Username must start with a letter
+        return "v" + hash
     }
 
-    let usernames = [];
-    for (var d in dids) {
-      if (!dids[d]) {
-        continue;
-      }
+    generateDatabaseName(did, contextName, databaseName) {
+        let text = [
+            did.toLowerCase(),
+            contextName,
+            databaseName,
+        ].join("/");
+        
+        const hash = EncryptionUtils.hash(text).substring(2);
+
+        // Database name must start with a letter
+        return "v" + hash
+    }
+
+    didsToUsernames(dids, contextName) {
+        if (!dids || !dids.length) {
+            return [];
+        }
+
+        let usernames = [];
+        for (var d in dids) {
+            if (!dids[d]) {
+                continue
+            }
+
+            const did = dids[d].toLowerCase()
+            usernames.push(this.generateUsername(did, contextName))
+        }
 
-      let did = dids[d].toLowerCase();
-      usernames.push(this.generateUsername(did, applicationName));
+        return usernames;
```

Comment:
Trying to understand why would you take `d in dids` and then again checking if d in dids 🤔 couldn't it just be something simpler like

```return dids?.map(did => this.generateUsername(did.toLowerCase(), contextName) ?? []```

Reply:
Also `did.toLowerCase()` happens inside the `this.generateUsername` so probably don't need it here.

Reply:
Agreed on all points. Updated.

Reply:
Had to change slightly to make it work due to current environment config.